packages/rust/server/tests/message_test.rs

```diff
@@ -0,0 +1,328 @@
+use crate::helper::{start_server, TempDatabase};
+
+use axum::http::StatusCode;
+
+use bubble::routes::message::{CheckMessages, MessageRequest, MessagesReturned};
+
+use uuid::Uuid;
+
+use std::str::FromStr;
+
+use bubble::routes::user::{Clients, CreateUser};
+use ed25519_dalek::{Keypair, PublicKey, SecretKey, Signer};
+use openmls::prelude::*;
+use openmls_rust_crypto::OpenMlsRustCrypto;
+
+use bubble::routes::client::CreateClient;
+
+use crate::crypto_helper::{generate_ed25519_keypair, PRIVATE, PUBLIC};
+use bubble::types::{Base64, SIGNATURE_SCHEME};
+
+mod crypto_helper;
+mod helper;
+
+#[tokio::test]
+async fn test_single_message() {
+    let db = TempDatabase::new().await;
+    let client = start_server(db.pool().clone()).await;
+
+    let created_user = CreateUser {
+        email: "test@gmail.com".to_string(),
+        username: "test_username".to_string(),
+        password: "test_password".to_string(),
+        name: "test_name".to_string(),
+        identity: Base64(PUBLIC.to_vec()),
+    };
+    let (token, _) = helper::initialize_user(db.pool(), &client, &created_user)
+        .await
+        .unwrap();
+
+    let bearer = format!("Bearer {}", token);
+    let (_, client_uuid) = helper::create_client(PUBLIC, PRIVATE, &bearer, &client).await;
+
+    let request_messages = CheckMessages {
+        client_uuid: client_uuid.to_string(),
+    };
+    let res = client
+        .get("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&request_messages).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+    assert_eq!(res.json::<MessagesReturned>().await.messages.len(), 0);
+
+    let message = MessageRequest {
+        client_uuids: vec![client_uuid.to_string()],
+        message: Base64("test message".as_bytes().to_vec()),
+    };
+    let res = client
+        .post("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&message).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+
+    let res = client
+        .get("/message")
+        .header("Content-Type", "application/json")
+        .body(serde_json::to_string(&request_messages).unwrap())
+        .header("Authorization", bearer.clone())
+        .send()
+        .await;
+
+    assert_eq!(res.status(), StatusCode::OK);
+    let messages = res.json::<MessagesReturned>().await.messages;
+    assert_eq!(messages.len(), 1);
+    assert_eq!("test message".as_bytes().to_vec(), messages[0].0);
```

Comment:
nit: store the message in a variable and then clone as needed

Reply:
i dont think this is necessary but I changed it for this test